sdk/eventhub/Azure.Messaging.EventHubs/tests/Primitives/EventProcessorTests.cs

```diff
@@ -22,6 +24,947 @@ namespace Azure.Messaging.EventHubs.Tests
     [TestFixture]
     public class EventProcessorTests
     {
+        /// <summary>
+        ///   Provides test cases for the constructor tests.
+        /// </summary>
+        ///
+        public static IEnumerable<object[]> ConstructorCreatesDefaultOptionsCases()
+        {
+            var connectionString = "Endpoint=sb://somehost.com;SharedAccessKeyName=ABC;SharedAccessKey=123;EntityPath=somehub";
+            var credential = Mock.Of<TokenCredential>();
+
+            yield return new object[] { new BasicProcessorMock(99, "consumerGroup", connectionString), "connection string with default options" };
+            yield return new object[] { new BasicProcessorMock(99, "consumerGroup", connectionString, default(EventProcessorOptions)), "connection string with explicit null options" };
+            yield return new object[] { new BasicProcessorMock(99, "consumerGroup", "namespace", "hub", credential), "namespace with default options" };
+            yield return new object[] { new BasicProcessorMock(99, "consumerGroup", "namespace", "hub", credential, default(EventProcessorOptions)), "namespace with explicit null options" };
```

Comment:
Should we be testing the constructor with a connection string and an event hub name as well? Also, should we directly test the `internal` constructor that takes a `PartitionLoadBalancer`? Tests seem to take only two constructors into consideration.

Reply:
By the way, testing both "default options" and "explicit null options" here won't make a difference because we are using the `BasicProcessorMock`. This class never calls the constructors with omitted options in the base class. We can add more constructors if we want to test that.

Reply:
I'm going to be honest with you here, I just ripped this out of the processor client tests and made minimal changes.   I didn't really give it much scrutiny.

Reply:
The default vs. explicit null doesn't really matter here, as we don't have any overloads that don't require the parameter.   We should test the hub name as well;  I back-ported to the client as well.